modules/http/http.c

```diff
@@ -282,6 +299,68 @@ http_dd_get_template_options(LogDriver *d)
   return &self->template_options;
 }
 
+void
+http_dd_set_ca_dir(LogDriver *d, const gchar *ca_dir)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  g_free(self->ca_dir);
+  self->ca_dir = g_strdup(ca_dir);
+}
+
+void
+http_dd_set_ca_file(LogDriver *d, const gchar *ca_file)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  g_free(self->ca_file);
+  self->ca_file = g_strdup(ca_file);
+}
+
+void
+http_dd_set_cert_file(LogDriver *d, const gchar *cert_file)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  g_free(self->cert_file);
+  self->cert_file = g_strdup(cert_file);
+}
+
+void
+http_dd_set_key_file(LogDriver *d, const gchar *key_file)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  g_free(self->key_file);
+  self->key_file = g_strdup(key_file);
+}
+
+void
+http_dd_set_cipher_suite(LogDriver *d, const gchar *ciphers)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  g_free(self->ciphers);
+  self->ciphers = g_strdup(ciphers);
+}
+
+void
+http_dd_set_peer_verify(LogDriver *d, const gchar *value)
+{
+  gboolean verify;
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  if (type_cast_to_boolean(value, &verify, NULL))
+    {
+      self->peer_verify = verify;
+    }
+  else
+    {
+      msg_warning("curl: non-boolean value for TLS peer verify",
+                  evt_tag_str("peer_verify", value));
+    }
+}
```

Comment:
The SSL syslog destination's peer-verify supports the following four values: optional-untrusted, optional-trusted, required-untrusted, required-trusted

I don't see a way to support the certificate being optional in https, at least not using the curl SSL/TLS options.  That is why I made it a binary switch.  If you think it is better to accept the same values as the syslog SSL destination and just treated the "optional" and "required" versions the same I'm happy to do that.

If you'd rather I used a different option name to avoid confusion, that would also work.

I didn't notice the yesno rules in the grammar, that would certainly be simpler.